presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java

```diff
@@ -170,6 +162,48 @@ else if (canQueueMore()) {
         }
     }
 
+    public ResourceGroupStateInfo getStateInfo()
+    {
+        synchronized (root) {
+            return new ResourceGroupStateInfo(
+                    id,
+                    getResourceGroupState(),
+                    new DataSize(softMemoryLimitBytes, BYTE),
+                    new DataSize(cachedMemoryUsageBytes, BYTE),
+                    maxRunningQueries,
+                    getAggregatedRunningQueriesInfo(),
+                    maxQueuedQueries,
+                    queuedQueries.size() + descendantQueuedQueries);
+        }
+    }
+
+    private ResourceGroupState getResourceGroupState()
```

Comment:
Maybe calling this getState is descriptive enough? The signature makes it pretty clear what's going on.